components/worktemplates/components/preview.tsx

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import styled from 'styled-components';
+
+import {AccordionItemType} from 'components/common/accordion/accordion';
+import {WorkTemplate} from '@mattermost/types/worktemplates';
+import {getTemplateDefaultIllustration} from '../utils';
+
+import ModalBodyWithIllustration from './modal_body_with_illustration';
+import Accordion from './preview/accordion';
+import Chip from './preview/chip';
+import PreviewSection from './preview/section';
+
+export interface PreviewProps {
+    className?: string;
+    template: WorkTemplate;
+}
+
+const Preview = ({template, ...props}: PreviewProps) => {
+    const {formatMessage} = useIntl();
+
+    const [currentIllustration, setCurrentIllustration] = React.useState<string>(getTemplateDefaultIllustration(template));
+
+    const channels = useMemo(() => template.content.filter((c) => c.channel).map((c) => c.channel!), [template.content]);
+    const boards = useMemo(() => template.content.filter((c) => c.board).map((c) => c.board!), [template.content]);
+    const playbooks = useMemo(() => template.content.filter((c) => c.playbook).map((c) => c.playbook!), [template.content]);
+    const integrations = useMemo(() => template.content.filter((c) => c.integration).map((c) => c.integration!), [template.content]);
```

Comment:
What about a single `for` to set all the required data (channels, boards, etc) ?